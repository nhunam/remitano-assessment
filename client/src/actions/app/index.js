import { ActionTypes } from "./type";
import { createActions } from "redux-actions";
//import proxy from 'http-proxy-middleware';

export const { clearMessage } = createActions({
  [ActionTypes.CLEAR_MESSAGE]: param => ({ ...param })
});
/*
export default function(app) {
  app.use(proxy('/users/**', { target: 'https://obscure-beach-46869.herokuapp.com/' }));
  app.use(proxy('/videos/**', { target: 'https://obscure-beach-46869.herokuapp.com/' }));
  app.use(proxy('/token/**', { target: 'https://obscure-beach-46869.herokuapp.com/' }));
};
*/