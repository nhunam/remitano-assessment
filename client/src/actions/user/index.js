import { ActionTypes } from "./type";
import { createActions } from "redux-actions";

export const {
  userLogin: login,
  userLogout: logout,
  userCreate: create
} = createActions({
  [ActionTypes.USER_LOGIN]: param => ({ ...param }),
  [ActionTypes.USER_CREATE]: param => ({ ...param }),
  [ActionTypes.USER_LOGOUT]: () => ({})
});
