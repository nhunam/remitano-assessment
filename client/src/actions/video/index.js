import { ActionTypes } from "./type";
import { createActions } from "redux-actions";

export const { videoFetch: fetch, videoSharing: shareVideo } = createActions({
  [ActionTypes.VIDEO_FETCH]: param => ({ ...param }),
  [ActionTypes.VIDEO_SHARING]: param => ({ ...param })
});
