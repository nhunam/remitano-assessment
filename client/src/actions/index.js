import { create, login, logout } from "./user";
import { fetch, shareVideo } from "./video";

import { clearMessage } from "./app";

export { login, logout, create, fetch, shareVideo, clearMessage };
