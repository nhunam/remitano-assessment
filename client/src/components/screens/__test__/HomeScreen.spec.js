import HomeScreen from "components/screens/Home";
import React from "react";
import { bindActionCreators } from "redux";
import { fetch } from "actions";
import { shallow } from "enzyme";

describe("Main Screen", () => {
  const mockDispatching = jest.fn();
  const fetchMock = bindActionCreators(fetch, mockDispatching);
  const videoMock = {
    data: [
      {
        id: 1,
        url: "test",
        title: "test",
        desription: "test",
        user: { id: 1, email: "test" }
      }
    ],
    page: {
      total_pages: 1,
      has_next: false,
      has_previous: false,
      current_page: 0,
      total_elements: 0
    }
  };
  const wrapper = shallow(<HomeScreen fetch={fetchMock} videos={videoMock} />);

  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should dispatch an fetch action when handle loadMore", () => {
    wrapper.instance().handleLoadMore(1);
    expect(mockDispatching).toHaveBeenCalledWith({
      payload: { page_index: 1 },
      type: "VIDEO_FETCH"
    });
  });
});
