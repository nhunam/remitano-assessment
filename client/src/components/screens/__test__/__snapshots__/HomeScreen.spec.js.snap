// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Screen should render properly 1`] = `ShallowWrapper {}`;
